### PR TITLE
Bug 1796301 - Add paddingEnd in OriginView for RTL config

### PR DESCRIPTION
**Pull request by rahulsainani**
### What
• Add `paddingEnd` for RTL layouts, we could add padding for LTR as well and set it directly in the layout, but the idea was to preserve the space in `OriginView` for the title/url.
• No tests since only RTL layout was affected.

### Screenshots

| Hebrew as RTL example | English as LTR example - unchanged | Desktop RTL |
|------|-----|----|
|Screenshot_20221206_141521|Screenshot_20221206_132722|Screen Shot 2022-12-07 at 10 45 34 AM|

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

File: android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
```
Comment (gabrielluong) [inline android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml:2]: Could you add the Mozilla license here? You can see an example with the `dimens.xml` file below.

File: android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+    <!-- DisplayToolbar -->
+    <dimen name="mozac_browser_toolbar_origin_padding_end">16dp</dimen>
```
Comment (gabrielluong) [inline android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml:4]: Just for my own curiosity, I am wondering how we came up with 16dp?

Comment (rahulsainani) [reply android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml:4]: Sure, it looked consistent with the other/icon paddings and at the start/end. Happy to update if there's a specific grid we adhere to.

File: android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml (hunk 1 above)
Comment (rahulsainani) [reply android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml:5]: Good catch, just added!

File: android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+</resources>
```
[the remark this answers is not in the document]
Comment (rahulsainani) [reply android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml]: Just added 👍

File: android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml (hunk 2 above)
Comment (gabrielluong) [reply android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml:7]: No worries, I think I just need a moment later today to pull the down the change and try to rationalize the padding of everything and see if the value fits with the existing layout. Normally, we would ask UX to double check if the screenshots look right, that said visually it looks right to me.

Comment (rahulsainani) [reply android-components/components/browser/toolbar/src/main/res/values-ldrtl/dimens.xml:7]: Sounds good! 👍

**Comment by gabrielluong** [comment]
@rahulsainani Just a quick status update, I am still digging through the Display Toolbar code, it's just been so long since I last looked at this and what I am kinda looking to piece together is how things are laid out and how do some of the views get their padding. It's not quite all in the XML and I am actually getting the sense that the approach might not necessarily be correct due to the lack of horizontal padding/margin in the XML. 

However, I also took a look at Desktop to see how does the RTL toolbar look, and I believe there's still more work to do if we actually wanted to implement RTL for the OriginView correctly. I am gonna ping some folks to weigh in.
Screen Shot 2022-12-07 at 10 45 34 AM

**Review by gabrielluong — CHANGES_REQUESTED** [review]
I am gonna request changes since I believe the correct solution to this is align the text to the right for RTL configurations.

**Comment by rahulsainani** [comment]
Thanks again for checking the desktop and mobile in RTL configs. Will wait for the UX feedback on whether to do it the fennick/chrome that you shared on slack (similar to this PR) or whether to make the text "aligned" to the right, for the lack of the better term.

**Review by gabrielluong — APPROVED** [review]
Spoke with @jonalmeida and @Amejia481 who're okay with the current approach and we'll make the best use of the situation to improve things in the meantime. We will look to follow up about seeing what the right look for RTL configuration should be after hearing back from Product/UX.

**Review by Amejia481 — APPROVED** [review]

**Pull request merged**